src/libsync/propagateuploadng.cpp

```diff
@@ -131,13 +310,18 @@ void PropagateUploadFileNG::slotPropfindFinished()
 
     _currentChunk = 0;
     _sent = 0;
-    while (_serverChunks.contains(_currentChunk)) {
-        _sent += _serverChunks[_currentChunk].size;
-        _serverChunks.remove(_currentChunk);
-        ++_currentChunk;
+
+    for (auto &chunkId : _serverChunks.keys()) {
+        if (updateRanges(chunkId, _serverChunks[chunkId].size)) {
+            _sent += _serverChunks[chunkId].size;
+            _serverChunks.remove(chunkId);
+        }
     }
 
-    if (_sent > _item->_size) {
+    if (!_rangesToUpload.isEmpty())
+        _currentChunk = _rangesToUpload.first().start;
```

Comment:
later: Potentially redundant, `_currentChunk` is also set in `startNextChunk()` - only used in logging, it seems.